app-libs/stf/src/trusted_call.rs

```diff
@@ -292,10 +295,12 @@ where
 	fn execute(
 		self,
 		calls: &mut Vec<ParentchainCall>,
+		shard: &ShardIdentifier,
 		node_metadata_repo: Arc<NodeMetadataRepository>,
 	) -> Result<(), Self::Error> {
 		let _role = ensure_authorization(&self)?;
 		// todo! spending limits according to role https://github.com/integritee-network/worker/issues/1656
+		ensure!(may_execute(&self), Self::Error::Filtered);
```

Comment:
we could filter upstream already, but it felt right to do it here because of DevX. Its nicer if customizations can happen in one place